pdi-execution-engine/api/src/main/java/org/pentaho/di/engine/api/IExecutionContext.java

```diff
@@ -1,17 +1,26 @@
 package org.pentaho.di.engine.api;
 
+import org.pentaho.di.engine.api.reporting.IProgressReporting;
+
 import java.util.Map;
+import java.util.concurrent.CompletableFuture;
 
 /**
  * Created by nbaker on 5/31/16.
  */
-public interface IExecutionContext {
-
+public interface IExecutionContext extends IProgressReporting {
   Map<String, Object> getParameters();
 
   Map<String, Object> getEnvironment();
 
   ITransformation getTransformation();
 
+  /**
+   * Transformation parameters are the preferred way of controlling execution.
+   * Maybe arguments can be rolled into environment?
+   */
+  @Deprecated
   String[] getArguments();
+
+  CompletableFuture<IExecutionResult> execute();
```

Comment:
@mkambol Execute was originally on IEngine, but I figured you would never want to execute on a context from a different engine, so this is safer. In the engines we were casting back into concrete implementations.

This could also allow allow for a builder-style fluent syntax `engine.prepare(someStuff).setParameters(yadaYada).execute() `